python/dgl/nn/pytorch/explain/subgraphxexplainer.py

```diff
@@ -0,0 +1,552 @@
+
+    Parameters
+    ----------
+    nodes: list
+        The node ids of the graph that are associated with this tree node.
+    pruning_action: str
+        A representation of the pruning action used to get to this node.
+    coef: float
+        The hyperparameter that encourages exploration, so high number encourages
+        exploration. It is high for moves with few simulations.
+    num_visit: int
+        Number times this node has been visited.
+    total_reward: float
+        The total reward for all node visits.
+    immediate_reward: float
+        Immediate reward for selecting this node (property score).
+    """
+
+    def __init__(
+        self,
+        nodes,
+        pruning_action,
+        coef=10.0,
+        num_visit=0,
+        total_reward=0.0,
+        immediate_reward=0.0,
+    ):
+        self.nodes = nodes
+        self.pruning_action = pruning_action
+        self.coef = coef
+        self.num_visit = num_visit
+        self.total_reward = total_reward
+        self.immediate_reward = immediate_reward
+        self.children = []
+
+    def average_reward(self):
+        r"""Get the average reward for multiple visits.
+
+        Returns
+        -------
+        float
+            Return the averaged reward for multiple visits.
+        """
+        return self.total_reward / self.num_visit if self.num_visit > 0 else 0
+
+    def action_selection_criteria(self, total_visit_count):
+        r"""Get the action selection criteria of node.
+
+        Parameters
+        ----------
+        total_visit_count: float
+            The total visiting counts for all possible actions of node.
+
+        Returns
+        -------
+        float
+            Returns the action selection criteria of node.
+        """
+        return (
+            self.coef
+            * self.immediate_reward
+            * math.sqrt(total_visit_count)
+            / (1 + self.num_visit)
+        )
+
+
+class SubgraphXExplainer(nn.Module):
+    r"""SubgraphXExplainer model from `SubgraphXExplainer: On Explainability of
+    Graph Neural Networks via Subgraph Explorations <https://arxiv.org/pdf/2102.05152.pdf>`.
+
+    It identifies the most important subgraph from the original graph that play a
+    critical role in GNN-based graph classification.
+
+    It employs Monte Carlo tree search in efficiently exploring different subgraphs for explanation
+    and uses Shapley values as the measure of subgraph importance.
+
+    Parameters
+    ----------
+    model: nn.Module
+        The GNN model to explain.
+    coef: float
+        The hyperparameter that encourages exploration, so high number encourages
+        exploration. It is high for moves with few simulations.
+    pruning_action: str
+        Pruning action either "High2low" or "Low2high" (refer to paper).
+        "High2low": Whether to expand children nodes from high degree to low degree when
+        extend the child nodes in the search tree. "Low2high" is opposite of "High2low".
```

Comment:
Since there are only two possibilities, use a bool argument `high2low` and set a default value `True`?

Reply:
done.